source/ExclusiveItem.h

```diff
@@ -0,0 +1,50 @@
+/* ExclusiveItem.h
+Copyright (c) 2022 by Amazinite
+
+Endless Sky is free software: you can redistribute it and/or modify it under the
+terms of the GNU General Public License as published by the Free Software
+Foundation, either version 3 of the License, or (at your option) any later version.
+
+Endless Sky is distributed in the hope that it will be useful, but WITHOUT ANY
+WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A
+PARTICULAR PURPOSE.  See the GNU General Public License for more details.
+*/
+
+#ifndef EXCLUSIVE_ITEM_H_
+#define EXCLUSIVE_ITEM_H_
+
+#include <memory>
+#include <utility>
+
+
+
+// An ExclusiveItem is an object stored as either a pointer or a stock
+// object, but not both.
+template <class Type>
+class ExclusiveItem {
+public:
+	ExclusiveItem() = default;
+
+	ExclusiveItem(const Type *item) : stockItem(item) {}
+	explicit ExclusiveItem(Type &&item) : item(std::move(item)) {}
+
+	ExclusiveItem(ExclusiveItem<Type> &&other) : stockItem(other.stockItem), item(std::move(other.item)) {}
```

Comment:
It seems like the copy and move constructor you defined do the same thing as the implicitly defined ones.

Reply:
From tehhowch: 
> [Please provide the rest of the special member functions (move & copy construct, move & copy assignment), even if it is just to default them.](https://github.com/endless-sky/endless-sky/blob/ee7d6ce2d6f0baa855b3c7cf70ec8ed289ccd0a7/source/Ship.cpp#L3948)

Reply:
`= default;` I think is what tehhowch meant.